spec.md

```diff
@@ -222,6 +222,29 @@ that contains both context and data).
   * OPTIONAL
 
 
+## Known extensions
```

Comment:
I'm not sure what it means to have a "known extension" in a spec. Is it a 2nd class field in some way? Is it less normative than the others?

If the intent is to define a set of fields that we think are good ideas but we're not yet prepared to make them part of the spec, I think it might be more appropriate to put them into a different doc so that its clear that they are experimental.  WDYT?

Reply:
I'll move this into a second doc. I think there is value in allowing parties to publicly document their extensions so that others can decide to follow suit and use the same meaning. Many standards, such as JWTs, iana media types, etc. do this.

I'm not 100% sure whether I'd ever push this to be a top-level field for a few reasons:
1. Not all services implement JWTs
2. Requiring a service to forward its JWT claims severely raises the bar since it requires more plumbing (and may not be acceptable to many)
3. Being an extension means there's a proto-safe way to distinguish between exists-but-empty (forwarding that a request was unauthenticated) and does not exist (the service does not forward JWT tokens or the request was authenticated a different way)

Reply:
+1
I like having a well known spot for the community to work together on extensions that might not ever become part of the spec, but still provides some level of interop value by just being well-defined some place.

Reply:
To be clear, I'm not commenting on whether or not this particular extension should be included - I'm more focused on creating the structure (secondary doc) in which we can place extensions in general.